### PR TITLE
Clean up CoffeeScript leftovers in webview code

### PR DESCRIPTION
**Pull request by kevinsawicki**
☕️ While working on #7852 I noticed lots of CoffeeScript :arrow_right: JavaScript leftovers in the code, i.e lots of unintended `return` statements, intermediate `ref` variables, etc which made the code hard to follow in certain places.

This pull request seeks to clean up that code since the `<webview>` implementation is one of the largest JavaScript pieces in the Electron codebase. Hopefully this makes it easier to work on going forward.

- [x] Use destructuring assignment
- [x] Use `let`/`const` instead of `var`
- [x] Use `for of` or `forEach` for array iteration
- [x] Use template strings
- [x] Use ES6 `class` syntax
- [x] Use `...` spread operator
- [x] Add more tests

### CODE REVIEW

File: lib/renderer/web-view/web-view-attributes.js [hunk 1]
```diff
@@ -141,18 +141,15 @@ class GuestInstanceAttribute extends WebViewAttribute {
...
 
   // Sets the attribute's value.
   setValue (value) {
     if (!value) {
-      return this.webViewImpl.webviewNode.removeAttribute(this.name)
-    }
-    if (isNaN(value)) {
-      return
+      this.webViewImpl.webviewNode.removeAttribute(this.name)
+    } else if (!isNaN(value)) {
```
Comment (zeke) [inline lib/renderer/web-view/web-view-attributes.js]: This could bite us:

```
> (!isNaN(null))
true
```

Maybe this would be better?

```js
Number(value) === value
```

File: lib/renderer/web-view/web-view.js [hunk 2]
```diff
@@ -407,38 +392,37 @@ var registerWebViewElement = function () {
...
+  for (const method of methods) {
+    proto[method] = createBlockHandler(method)
   }
-  createNonBlockHandler = function (m) {
+
+  const createNonBlockHandler = function (m) {
     return function (...args) {
       const internal = v8Util.getHiddenValue(this, 'internal')
-      return ipcRenderer.send.apply(ipcRenderer, ['ELECTRON_BROWSER_ASYNC_CALL_TO_GUEST_VIEW', null, internal.guestInstanceId, m].concat(args))
+      ipcRenderer.send('ELECTRON_BROWSER_ASYNC_CALL_TO_GUEST_VIEW', null, internal.guestInstanceId, m, ...args)
```
Comment (zeke) [inline lib/renderer/web-view/web-view.js]: This is so much better than those `apply`/`concat` shenanigans. 👍

File: spec/webview-spec.js [hunk 3]
```diff
@@ -1256,4 +1254,40 @@ describe('<webview> tag', function () {
...
+
+      webview.addEventListener('resize', function onResize (event) {
+        webview.removeEventListener('resize', onResize)
+        assert.equal(event.newWidth, 1234)
+        assert.equal(event.newHeight, 789)
+        assert.equal(event.target, webview)
+        done()
+      })
+
+      webview.src = 'file://' + fixtures + '/pages/a.html'
```
Comment (zeke) [inline spec/webview-spec.js]: Could be a template string, but not important.

File: lib/browser/guest-view-manager.js [hunk 4]
```diff
@@ -160,24 +151,24 @@ const attachGuest = function (embedder, elementInstanceId, guestInstanceId, para
     destroyGuest(embedder, oldGuestInstanceId)
   }
 
-  guestInstance = guestInstances[guestInstanceId]
+  const guestInstance = guestInstances[guestInstanceId]
   // If this isn't a valid guest instance then do nothing.
   if (!guestInstance) {
     return
```
Comment (kevinsawicki) [inline lib/browser/guest-view-manager.js]: Yeah, I wasn't planning on touching these, I'd be up for doing them in a secondary pass though

File: lib/browser/guest-view-manager.js [hunk 5]
```diff
@@ -280,12 +271,12 @@ ipcMain.on('ELECTRON_GUEST_VIEW_MANAGER_DESTROY_GUEST', function (event, guestIn
...
-  return guestInstance != null ? guestInstance.guest.setSize(params) : void 0
+  const guest = getGuest(guestInstanceId)
+  if (guest != null) guest.setSize(params)
 })
 
 // Returns WebContents from its guest id.
-exports.getGuest = function (guestInstanceId) {
+const getGuest = function (guestInstanceId) {
   const guestInstance = guestInstances[guestInstanceId]
   return guestInstance != null ? guestInstance.guest : void 0
```
Comment (kevinsawicki) [inline lib/browser/guest-view-manager.js]: Yeah, `void 0` is a CoffeeScript thing, will update this with your suggestion 👍

File: lib/renderer/web-view/web-view-attributes.js (hunk 1 above)
Comment (kevinsawicki) [inline lib/renderer/web-view/web-view-attributes.js]: I was hope to not change any behavior in this pull request, would be up for doing this in a follow on where we get a failing spec first and then update it 👍

**Comment by kevinsawicki** [comment]
Build failures are unrelated to this change, looks like timeouts that are also occurring on master, shipping this, 🚢

Thanks @zeke for reviewing this thoroughly 👀
